### PR TITLE
Add statsd metrics

### PR DESCRIPTION
**Pull request by Natim**
- [x] Time calls to the TokenServer (SyncClient)
- [x] Time calls to the SyncServer (SyncClient methods)
- [x] Time calls to the cache service (already provided by Cliquet)

### CODE REVIEW

File: syncto/authentication.py [hunk 1]
```diff
@@ -48,8 +48,20 @@ def build_sync_client(request):
     if not credentials:
         ttl = int(settings['syncto.cache_credentials_ttl_seconds'])
         tokenserver = TokenserverClient(bid_assertion, client_state)
+        if request.registry.statsd:
```
Comment (leplatrem) [inline syncto/authentication.py]: I would recommend a local variable `statsd` to avoid repeating

File: syncto/authentication.py [hunk 2]
```diff
@@ -48,8 +48,20 @@ def build_sync_client(request):
...
+            request.registry.statsd.watch_execution_time(tokenserver,
+                                                         prefix="tokenserver")
         credentials = tokenserver.get_hawk_credentials(duration=ttl)
         request.registry.cache.set(cache_key, credentials, ttl)
 
+    if request.registry.statsd:
```
Comment (leplatrem) [inline syncto/authentication.py]: You can also add a default setting for statsd to avoid testing if it's setup, no ?

**Comment by leplatrem** [comment]
I am not an expert of statsd but it looks alright !

File: syncto/authentication.py (hunk 2 above)
Comment (Natim) [inline syncto/authentication.py]: You means defaulting to a Dummy statsd driver if none is configured?

**Pull request merged**